modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SFor.java

```diff
@@ -52,12 +52,15 @@ public SFor(Location location, ANode initializer, AExpression condition, AExpres
     }
 
     @Override
-    void analyze(ScriptRoot scriptRoot, Scope scope) {
+    Output analyze(ScriptRoot scriptRoot, Scope scope, Input input) {
+        this.input = input;
+        output = new Output();
```

Comment:
Same as above, consider changing where this creation occurs.  It's about ~50 lines away from actual use.

Reply:
Done.